### ISSUE TITLE
[Bug] 3d model slower after 8-bit quantization on iGPU

### ISSUE THREAD

**Issue by hyperfraise** [openvinotoolkit/openvino#4887]
##### System information (version)

- OpenVINO => 2021.2.185
- Operating System / Platform => Ubuntu 18
- Problem classification: Model Quantization
- Framework: Onnx
- Model name: ResNet18
- Hardware : Intel i3-7200U (Intel HD 620)

##### Detailed description
My 3d model is slower after quantization on GPU (but indeed faster on CPU).

##### Steps to reproduce

I'm attaching here the quantized model and the fp16 model. 

[int8_resnet.zip](https://github.com/openvinotoolkit/openvino/files/6171624/int8_resnet.zip)
[fp16_resnet.zip](https://github.com/openvinotoolkit/openvino/files/6171385/fp16_resnet.zip)
 
And here are the scripts to generate them an the onnx file as well : 

First run this to export the onnx model.
```
import torch
import torchvision

dummy_input = torch.randn(1, 3, 35, 224, 224)
model = torchvision.models.video.r3d_18()

torch.onnx.export(
    model,
    dummy_input,
    "resnet.onnx",
    verbose=True,
)
```

Then I enter the official container for openvino : 
`docker run --name openvino -it --device /dev/dri:/dev/dri --device-cgroup-rule='c 189:* rmw' -v /dev/bus/usb:/dev/bus/usb --rm openvino/ubuntu18_dev:latest`

Then I run the conversion script : 
```
cd /opt/intel/openvino_2021.2.185/deployment_tools/model_optimizer
python3 mo.py --input_model /home/openvino/resnet.onnx  --output_dir /home/openvino/ --data_type FP16
```

Then I run the following quantization script with dummy dataset (I don't care about accuracy, just speed) : 
```
import os

import nibabel as nib
import numpy as np
from scipy.ndimage import interpolation
from addict import Dict

from compression.api import Metric, DataLoader
from compression.engines.ie_engine import IEEngine
from compression.graph import load_model, save_model
from compression.graph.model_utils import compress_model_weights
from compression.pipeline.initializer import create_pipeline
from compression.utils.logger import init_logger
from sample.utils.argument_parser import get_common_argparser

# Initialize the logger to print the quantization process in the console.
init_logger(level="DEBUG")


# Custom DataLoader class implementation that is required for
# the proper reading of BRATS 3D Segmentation images and annotations.
class DummyDatasetLoader(DataLoader):

    # Required methods:
    def __init__(self, config):
        if not isinstance(config, Dict):
            config = Dict(config)
        super().__init__(config)

    def __getitem__(self, index):
        """
        Returns annotation, image and image metadata at the specified index.
        Possible formats:
        (img_id, img_annotation), image
        (img_id, img_annotation), image, image_metadata
        """

        if index >= len(self):
            raise IndexError

        return (
            [index, np.random.choice(10)],
            np.random.choice(range(256), size=(3, 35, 224, 224)).astype(np.uint8),
            {"image_shape": (3, 35, 224, 224)},
        )

    def __len__(self):
        """ Returns size of the dataset """
        return 10


# raw output of the model using metadata obtained during the image reading and preprocessing.
class SegmentationEngine(IEEngine):
    @staticmethod
    def postprocess_output(outputs, metadata):
        """
        Processes model raw output for future metric and loss calculation.
        Uses image metadata that can be passed using dataloader.
        :param outputs: network infer result in format of numpy ndarray (batch x image shape)
        :param metadata: dictionary of image metadata
        :return: processed numpy ndarray with the same shape as the original output
        """
        return outputs


def main():
    parser = get_common_argparser()

    args = parser.parse_args()
    if not args.weights:
        args.weights = "{}.bin".format(os.path.splitext(args.model)[0])

    model_config = Dict(
        {
            "model_name": "resnet",
            "model": os.path.expanduser(args.model),
            "weights": os.path.expanduser(args.weights),
        }
    )

    engine_config = Dict(
        {"device": "GPU", "stat_requests_number": 4, "eval_requests_number": 4}
    )

    dataset_config = Dict(
        {
            "modality_order": [1, 2, 3, 0],
            "size": (35, 224, 224),
        }
    )

    algorithms = [
        {
            "name": "DefaultQuantization",
            "params": {
                "target_device": "GPU",
                "preset": "performance",
                "stat_subset_size": 100,
            },
        }
    ]

    # Step 1: Load the model.
    print(model_config)
    model = load_model(model_config)
    # model = ie.read_network(
    #     model=model_config["model"], weights=model_config["weights"]
    # )
    # # Loading the network to the inference engine
    # model = ie.load_network(network=model, device_name="GPU")

    # Step 2: Initialize the data loader.
    data_loader = DummyDatasetLoader(dataset_config)

    # Step 4: Initialize the engine for metric calculation and statistics collection.
    engine = SegmentationEngine(config=engine_config, data_loader=data_loader)

    # Step 5: Create a pipeline of compression algorithms.
    pipeline = create_pipeline(algorithms, engine)

    # Step 6: Execute the pipeline.
    compressed_model = pipeline.run(model)

    # Step 7 (Optional):  Compress model weights to quantized precision
    #                     in order to reduce the size of final .bin file.
    compress_model_weights(compressed_model)

    # Step 8: Save the compressed model to the desired path.
    save_model(compressed_model, os.path.join(os.path.curdir, "optimized"))

    # Step 9 (Optional): Evaluate the compressed model. Print the results.
    metric_results = pipeline.evaluate(compressed_model)
    if metric_results:
        for name, value in metric_results.items():
            print("{: <27s}: {}".format(name, value))


if __name__ == "__main__":
    main()
```
I run it with these parameters : 
```
cd
python3 quantize_model.py --model resnet.xml --dataset none
```

And finally, here is my small script for speedtest : 
```

from openvino.inference_engine import IECore

model_bin = "resnet.bin"
# Loading the Inference Engine API
ie = IECore()
ie.available_devices
# Loading IR files
net = ie.read_network(model="resnet.xml", weights=model_bin)
# Loading the network to the inference engine
exec_net = ie.load_network(network=net, device_name="GPU")
print("IR successfully loaded into Inference Engine.")

import cv2
import numpy as np
batch_size = 1
video = np.random.choice(range(256), size=(batch_size, 3, 35, 224, 224)).astype(
    np.uint8
)
input_blob = next(iter(exec_net.inputs))
exec_net.infer({input_blob: video})
import time
t = time.time()
num_iterations = 10
for i in range(num_iterations):
    _ = exec_net.infer({input_blob: video})
print("Speed (spl/s) : ", (num_iterations * batch_size) / (time.time() - t))
```

That I run with no argument. For reproduction, don't forget to adapt the model name.

It gives me such results : 
```
On iGPU :
FP32 model : 
Speed (spl/s) :  0.43001067946492383
FP16 model : 
Speed (spl/s) :  0.7477045227425639
Int8 model : 
Speed (spl/s) :  0.6828410222974903
```

As I noted above, the INT8 speed **on CPU** is as expected ~2 times faster than FP16. And notably FP16 speed is the same as FP32 (although I don't really care about that).
```
On CPU : 
FP32 model : 
Speed (spl/s) :  0.2582671660972424
FP16 model : 
Speed (spl/s) :  0.2582276659586155
INT8 model : 
Speed (spl/s) :  0.45292130833715755
```

What exactly is happening wrong ? And more importantly : can I expect a speedup on different iGPUs ? For example will the INT8 speed be slower than FP16 on 1165G7, or will it be about ~2 times faster than FP16 on it as I would expect ? 

Is it purely related to 3d operators implementation in openvino on iGPU, meaning I won't ever have a INT8 speedup on iGPU until openvino is updated to improve the 3d operators support for INT8 on iGPU ? In which case, is there a plan to add support for those ?

**Comment by hyperfraise** [comment]
Note that if I quantize an FP32 model on iGPU, I end up with a model slower than FP32 !

**Comment by vladimir-paramuzov** [comment]
@hyperfraise Intel HD 620 GPU doesn't have hardware acceleration for int8 execution, and GPU plugin disables low precision transformations for such cases, i.e. the model is executed in the original precision (FP32 in your case). But since the quantized model contains a lot of additional FakeQuantize operations, the performance could be a bit worse comparing to original non-quantized model. 
For 1165G7 with Iris Xe GPU you might expect speedup for int8 model

**Comment by hyperfraise** [comment]
Thanks.

**Comment by hyperfraise** [comment]
Hi. I

**Comment by vladimir-paramuzov** [comment]
@hyperfraise Low precision models in OpenVINO are stored as original FP32/16 model with bunch of FakeQuantize operations which don't change the actual operators precision in IR. These FakeQuantize operations are transformed on model loading stage to get actual int8 operations in the graph.

**Comment by hyperfraise** [comment]
Ok. Well i'm a bit disappointed by the speed. With d networks I see an almost 95% speedup, and only 40% here. losing this.
